core-library/src/main/java/org/silverpeas/core/contribution/template/publication/PublicationTemplateManager.java

```diff
@@ -510,4 +527,136 @@ protected void registerForRenewingContentCipher() {
+          "Context is not complete ! ComponentId, ObjectId, Language and UserId must be " +
+              "defined...");
+    }
+    if (StringUtil.isDefined(xmlFormName)) {
+      PublicationTemplate pub = getTemplateAndRegisterIt(xmlFormName, context);
+      try {
+        RecordSet set = pub.getRecordSet();
+        Form form = pub.getUpdateForm();
+        DataRecord data = set.getRecord(context.getObjectId());
+        if (data == null) {
+          data = set.getEmptyRecord();
+          data.setId(context.getObjectId());
+        }
+
+        // sauvegarde des données du formulaire
+        form.update(items, data, context);
+        set.save(data);
+      } catch (Exception e) {
+        throw new SilverpeasException("Unable to save data", e);
+      }
+    }
+  }
+
+  private PublicationTemplate getTemplateAndRegisterIt(String xmlFormName, PagesContext context)
+      throws SilverpeasException {
+    String shortName = getShortName(xmlFormName);
+    String externalId = context.getComponentId() + ":" + shortName;
+    PublicationTemplate pub;
+    try {
+      // récupération des données du formulaire (via le DataRecord)
+      pub = getPublicationTemplate(externalId);
+    } catch (Exception e) {
+      try {
+        addDynamicPublicationTemplate(externalId, shortName);
+        pub = getPublicationTemplate(externalId);
+      } catch (Exception templateException) {
+        throw new SilverpeasException("Unable to register template", templateException);
+      }
+    }
+    return pub;
+  }
+
+  public Form getFormAndData(String xmlFormName, PagesContext context, boolean readOnly)
+      throws SilverpeasException {
+    if (StringUtil.isNotDefined(xmlFormName)) {
+      return null;
+    }
+    PublicationTemplate pub = getTemplateAndRegisterIt(xmlFormName, context);
+    try {
+      RecordSet set = pub.getRecordSet();
+      Form form = pub.getUpdateForm();
+      if (readOnly) {
+        form = pub.getViewForm();
+      }
+      DataRecord data = set.getRecord(context.getObjectId());
+      if (data == null) {
+        if (readOnly) {
+          return null;
+        }
+        data = set.getEmptyRecord();
+        data.setId(context.getObjectId());
+      }
+      form.setData(data);
+      return form;
+    } catch (Exception e) {
+      throw new SilverpeasException("Unable to get data", e);
+    }
+  }
+
+  public void setDataIntoIndex(String xmlFormName, String componentId, String userId,
+      FullIndexEntry indexEntry) {
+    try {
+      String shortName = getShortName(xmlFormName);
+      PublicationTemplate usedTemplate = getPublicationTemplate(componentId + ":" + shortName);
+      RecordSet set = usedTemplate.getRecordSet();
+      set.indexRecord(userId, shortName, indexEntry);
+    } catch (Exception e) {
+      SilverLogger.getLogger(this).error(e);
+    }
+  }
+
+  public PublicationTemplate getDirectoryTemplate() {
+    List<PublicationTemplate> templates = getDirectoryTemplates();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "templates" which hides the field declared at line 89. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)